protostar/commands/init/project_creator/new_project_creator.py

```diff
@@ -52,12 +66,82 @@ def _gather_input(self) -> "NewProjectCreator.NewProjectConfig":
-        self.save_protostar_toml(
-            project_root_path=project_root_path, cairo_version=cairo_version
+        project_root_path = output_dir_path / project_config.project_dirname
+        project_name = project_config.project_dirname
+
+        self.create_project_directory_from_template(
+            cairo_version=cairo_version,
+            project_root_path=project_root_path,
+            project_name=project_name,
+        )
+        self._write_protostar_toml_from_config(
+            project_root_path=project_root_path,
+            configuration_model=self._new_project_config(cairo_version=cairo_version),
+        )
+
+    def create_project_directory_from_template(
+        self, cairo_version: CairoVersion, project_root_path: Path, project_name: str
+    ):
+        template_path = self.script_root / "templates" / cairo_version.value
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_dir = Path(temp_dir)
+            new_template_path = Path(temp_dir) / "template"
+
+            shutil.copytree(template_path, new_template_path)
+            self._insert_project_name_to_template(project_name, new_template_path)
+
+            try:
+                shutil.copytree(new_template_path, project_root_path)
+            except FileExistsError as ex_file_exists:
+                raise ProtostarException(
+                    f"Folder or file named {project_root_path.name} already exists. Choose different project name."
+                ) from ex_file_exists
+
+    def _new_project_config(
+        self, cairo_version: CairoVersion
+    ) -> ConfigurationFileV2Model:
+        if cairo_version == CairoVersion.cairo0:
+            return ConfigurationFileV2Model(
+                protostar_version=str(self._protostar_version),
+                contract_name_to_path_strs={"main": ["src/main.cairo"]},
+                project_config={
+                    "lib-path": "lib",
+                },
+                command_name_to_config={},
+                profile_name_to_project_config={},
+                profile_name_to_commands_config={},
+            )
+
+        return ConfigurationFileV2Model(
+            protostar_version=str(self._protostar_version),
+            contract_name_to_path_strs={"hello_starknet": ["src"]},
+            project_config={
+                "lib-path": "lib",
+                "linked-libraries": ["src"],
+            },
         )
+
+    @staticmethod
+    def _insert_project_name_to_template(project_name: str, temp_dir: Path):
+        for path in temp_dir.glob("**/*"):
+            if not path.is_file():
+                continue
+
+            file_contents = path.read_text("utf-8")
+            file_contents = Template(file_contents).substitute(
+                PROJECT_NAME=project_name
+            )
+            path.write_text(file_contents, encoding="utf-8")
+
+    @staticmethod
+    def _validate_project_name(name: str):
```

Comment:
I would add a comment/rename the function to suggest that it is done because of Scarb